### PR TITLE
docs(monitoring): document grafana and influxdb persistence

### PR DESCRIPTION
**Pull request by rimusz**

### CODE REVIEW

**Comment by rimusz** [comment]
@vdice can you take a look to the updated doc and try to set persistence for grafana and influxdb

File: src/managing-workflow/platform-monitoring.md [hunk 1]
```diff
@@ -52,18 +52,30 @@ Grafana will preload several dashboards to help operators get started with monit
...
-Deis Workflow monitoring does not currently write data to the host filesystem or to long-term storage. If the Grafana
-instance fails, modified dashboards are lost. Until there is a solution to persist this, export dashboards and store
-them separately in version control.
+Deis Workflow monitoring by default does not write data to the host filesystem or to long-term storage. If the Grafana instance fails, modified dashboards are lost. 
+
+### On Cluster Persistence
+
+If you wish to have persistence for Grafana you can set `enabled` to `true` in the `values.yaml` file before running `helm install`.
+
+* ` grafana:`
```
Comment (bacongobbler) [inline src/managing-workflow/platform-monitoring.md]: The formatting here is odd. It's a bullet point list with embedded code. Can you change it to a code snippet, like so?

```
grafana:
  # Configure the following ONLY if you want persistence for on-cluster grafana
  # GCP PDs and EBS volumes are supported only
  persistence:
    enabled: true
    size: 5Gi
```

File: src/managing-workflow/platform-monitoring.md [hunk 2]
```diff
@@ -75,6 +87,20 @@ You can choose to not expose the Influx UI and API to the world by updating
 `$CHART_HOME/workspace/workflow-$WORKFLOW_RELEASE/manifests/deis-monitor-influxdb-ui-svc.yaml` and removing the
 following line - `router.deis.io/routable: "true"`.
 
+### On Cluster Persistence
+
+If you wish to have persistence for InfluxDB you can set `enabled` to `true` in the `values.yaml` file before running `helm install`.
+
+* ` influxdb:`
```
Comment (bacongobbler) [inline src/managing-workflow/platform-monitoring.md]: same here with the wonky formatting

File: src/managing-workflow/platform-monitoring.md [hunk 3]
```diff
@@ -52,18 +52,30 @@ Grafana will preload several dashboards to help operators get started with monit
...
 ### Off Cluster Grafana
 
-It is recommended that users provide their own installation for Grafana if possible. The current deployment of Grafana within Workflow is not durable across pod restarts which means custom dashboards that are created after startup will not be restored when the pod comes back up. If you wish to provide your own Grafana instance you can set `grafana_location` in the `values.yaml` file before running `helm install`.
+If you wish to provide your own Grafana instance you can set `grafana_location` in the `values.yaml` file before running `helm install`.
 
 ## InfluxDB
 
 InfluxDB writes data to the host disk, however, if the InfluxDB pod dies and comes back on
-another host the data will not be recovered. We intend to fix this in a future release. The InfluxDB Admin UI is also
+another host the data will not be recovered, you need to enable on-cluster persistence for data to persist. The InfluxDB Admin UI is also
```
Comment (bacongobbler) [inline src/managing-workflow/platform-monitoring.md]: probably better to switch the comma for a period here, as they are two complete sentences.

> if the InfluxDB pod dies and comes back on another host, the data will not be recovered. You need to enable on-cluster persistence for data to persist.

File: src/managing-workflow/platform-monitoring.md [hunk 4]
```diff
@@ -85,6 +111,41 @@ To use off-cluster Influx, please provide the following values in the `values.ya
 * `user = "InfluxUser"`
 * `password = "MysuperSecurePassword"`
 
+
+## PVC Dynamic Provisioning
+
+Kubernetes v1.4.x has introduced Dynamic Provisioning and Storage Classes, you can read about it [here](http://blog.kubernetes.io/2016/10/dynamic-provisioning-and-storage-in-kubernetes.html).
```
Comment (bacongobbler) [inline src/managing-workflow/platform-monitoring.md]: V in V1.4.x should be lowercase for consistency.

File: src/managing-workflow/platform-monitoring.md [hunk 5]
```diff
@@ -75,6 +87,20 @@ You can choose to not expose the Influx UI and API to the world by updating
...
+If you wish to have persistence for InfluxDB you can set `enabled` to `true` in the `values.yaml` file before running `helm install`.
+
+* ` influxdb:`
+* `   # Configure the following ONLY if you want persistence for on-cluster grafana`
+* `   # GCP PDs and EBS volumes are supported only`
+* `   persistence:`
+* `     enabled: true # Set to true to enable persistence`
+* `     size: 5Gi # PVC size`
+
+You have to set (if you do not have it already) `standard` StorageClass as per [PVC Dynamic Provisioning](#pvc-dynamic-provisioning), as it does not get set by default in Kubernetes v1.4.x and v1.5.x.
```
Comment (bacongobbler) [inline src/managing-workflow/platform-monitoring.md]: V in V1.4.x and V1.5.x should be lowercase for consistency.

File: src/managing-workflow/platform-monitoring.md [hunk 6]
```diff
@@ -52,18 +52,30 @@ Grafana will preload several dashboards to help operators get started with monit
...
+If you wish to have persistence for Grafana you can set `enabled` to `true` in the `values.yaml` file before running `helm install`.
+
+* ` grafana:`
+* `   # Configure the following ONLY if you want persistence for on-cluster grafana`
+* `   # GCP PDs and EBS volumes are supported only`
+* `   persistence:`
+* `     enabled: true # Set to true to enable persistence`
+* `     size: 5Gi # PVC size`
+
+You have to set (if you do not have it already) `standard` StorageClass as per [PVC Dynamic Provisioning](#pvc-dynamic-provisioning), as it does not get set by default in Kubernetes v1.4.x and v1.5.x.
```
Comment (bacongobbler) [inline src/managing-workflow/platform-monitoring.md]: V in V1.4.x and V1.5.x should be lowercase for consistency.

Comment (rimusz) [inline src/managing-workflow/platform-monitoring.md]: v is lowercase there already

File: src/managing-workflow/platform-monitoring.md (hunk 5 above)
Comment (rimusz) [inline src/managing-workflow/platform-monitoring.md]: v is lowercase there already

File: src/managing-workflow/platform-monitoring.md (hunk 4 above)
Comment (rimusz) [inline src/managing-workflow/platform-monitoring.md]: v is lowercase there already

File: src/managing-workflow/platform-monitoring.md (hunk 6 above)
Comment (bacongobbler) [inline src/managing-workflow/platform-monitoring.md]: I'm having a hard time reading today. Sorry

**Comment by rimusz** [comment]
@bacongobbler updated as per your comments

File: src/managing-workflow/platform-monitoring.md [hunk 7]
```diff
@@ -85,6 +115,41 @@ To use off-cluster Influx, please provide the following values in the `values.ya
...
+
+## PVC Dynamic Provisioning
+
+Kubernetes v1.4.x has introduced Dynamic Provisioning and Storage Classes, you can read about it [here](http://blog.kubernetes.io/2016/10/dynamic-provisioning-and-storage-in-kubernetes.html).
+
+To use persistence for Grafana and InfluxDB you also need to deploy StorageClass objects to the Kubernetes cluster with `kubectl create -f storage-standard.yaml`.
+
+Note: AWS and GCE/GKE have different `StorageClass` settings.
+
+AWS `storage-standard.yaml` manifest:
```
Comment (vdice) [inline src/managing-workflow/platform-monitoring.md]: It looks like this should be the `GCE/GKE` variant and the one below `AWS`...

**Comment by rimusz** [comment]
@vdice good spotting, updated

**Pull request merged**